scripts/create_backend_sandbox.sh

```diff
@@ -0,0 +1,72 @@
+#!/bin/bash
+
+set -euo pipefail
+
+# expect 1 argument to the script
+if [ $# -ne 1 ]; then
+  echo 1>&2 "$0: expected 1 arguments, received $#"
+  exit 2
+fi
+
+ENVIRONMENT=$1
+COMPONENT="sandbox"
+AWS_ACCOUNT_ID="$(aws sts get-caller-identity --query Account --output text)"
+AWS_REGION="eu-west-2"
+PROJECT="nhs-notify"
+GROUP="nhs-notify-template-management-nonprod"
+
+root_dir=$(git rev-parse --show-toplevel)
+terraform_dir=$root_dir/infrastructure/terraform
+component_dir=$terraform_dir/components/$COMPONENT
+
+
+echo "Creating backend sandbox \"$ENVIRONMENT\""
+
+cd $terraform_dir
+
+./bin/terraform.sh \
+  --project $PROJECT \
+  --region $AWS_REGION \
+  --component $COMPONENT \
+  --environment $ENVIRONMENT \
+  --group $GROUP \
+  --action apply \
+  -- \
+  -var aws_account_id=$AWS_ACCOUNT_ID \
+  -var region=$AWS_REGION \
+  -var project=$PROJECT \
+  -var environment=$ENVIRONMENT \
+  -var group=$GROUP
+
+cd $component_dir
+
+# This is lifted from infrastructure/terraform/bin/terraform.sh
+# That script creates and deletes its own terraform backend configuration, and then deletes it again
+# We can run the `output` command with the terraform.sh script, but can't write the output to a file
+# because it prints loads of other things to stdout too
+# so we need to reconfigure the backend and run the init/output commands using the terraform binary directly
+
+# create tf backend config
+bucket="${PROJECT}-tfscaffold-${AWS_ACCOUNT_ID}-${AWS_REGION}"
+backend_prefix="${PROJECT}/${AWS_ACCOUNT_ID}/${AWS_REGION}/${ENVIRONMENT}";
+backend_filename="${COMPONENT}.tfstate";
+
+readonly backend_key="${backend_prefix}/${backend_filename}";
+readonly backend_config="terraform {
+  backend \"s3\" {
+    region         = \"${AWS_REGION}\"
+    bucket         = \"${bucket}\"
+    key            = \"${backend_key}\"
+    dynamodb_table = \"${bucket}\"
+  }
+}";
+
+# write the backend file
+echo -e "${backend_config}" > backend_tfscaffold.tf
+
+# clean up the file on exit
+trap "rm -f $(pwd)/backend_tfscaffold.tf" EXIT;
+
+terraform init -upgrade
```

Comment:
what does the extra init do?

Reply:
Let me try it without 😅

Reply:
Yup, you are right. The problem was that the `terraform.sh` script creates and then deletes the backend config, so I had to manually create that again. But before I'd done that, the error I was seeing said about running `init` again. So I just added that in without really thinking it through. Thanks.